client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
+
+export const requestUserSettings = ( { dispatch }, action, next ) => {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/me/settings',
+	}, action ) );
+
+	return next( action );
+};
+
+export const receiveUserSettings = ( { dispatch }, action, next, data ) => {
+	dispatch( updateUserSettings( fromApi( data ) ) );
+};
+
+export const receiveRequestError = noop;
+
+/*
+ * Post settings to WordPress.com API at /me/settings endpoint
+ */
+export function saveUserSettings( { dispatch, getState }, action, next ) {
+	const { settingsOverride } = action;
+	const settings = settingsOverride || getState().userSettings.unsavedSettings;
```

Comment:
There are two ways how to work with this module in the UI:
1. Present a form where the user can change one or more values and then hit "Submit". In this case, we dispatch `updateUserSetting` on every field change and one `saveUserSettings` without parameters on submit. The intermediate unsaved state is stored in `unsavedSettings`.
2. Present a control that changes certain setting immediately without any intermediate workflow. For example, we will have a one-click language switcher in the sidebar soon. Such a control will directly dispatch `saveUserSettings` with just one changed value. And it will not disrupt the unsaved form that might me displayed on the same screen.

It might not be ideal API, but it's how the current `client/lib/user-settings` module works and it works well in all places where it's used in Calypso.

I'll use the `getUnsavedSettings` selector. Also, I'll rename the `updateUserSetting` action to `setUserSetting`. When renaming the `receive*` actions to `update*`, I didn't notice that I created a confusing name conflict - the actions for altering a setting and for receiving settings from server both use the `update` verb.